### ISSUE TITLE
[Bug]: Fatal Exception: java.lang.RuntimeException

### ISSUE THREAD

**Issue by Petr-Kubista** [OneSignal/OneSignal-Android-SDK#1867]
### What happened?

I am getting some reports in Crashlytics from Samsung users with Android 11. Unfortunately unable to reproduce:

```
Fatal Exception: java.lang.RuntimeException: Unable to create application

Caused by org.json.JSONException: No value for notification_types
       at org.json.JSONObject.get(JSONObject.java:400)
       at org.json.JSONObject.getInt(JSONObject.java:489)
       at com.onesignal.internal.OneSignalImp.initWithContext(OneSignalImp.kt:224)
       at com.onesignal.OneSignal.initWithContext(OneSignal.kt:126)
       at org.xxx.MyApplication.initPushNotifications(MyApplication.java:157)
       at org.xxx.MyApplication.onCreate(MyApplication.java:132)
```

### Steps to reproduce?

```Markdown
Unable to reproduce
```

### What did you expect to happen?

App starting normally without a crash

### OneSignal Android SDK version

5.0.2

### Android version

11

### Specific Android models

```Markdown
* Samsung Galaxy A71
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
Thank you for reporting! We're looking into this and will get back to you as soon as possible.

**Comment by area51bis** [comment]
I think it can be fixed by changing the line OneSignalImp.kt:280:

`val notificationTypes = legacyUserSyncJSON.getInt("notification_types")`

por:

`val notificationTypes = legacyUserSyncJSON.optInt("notification_types", SubscriptionStatus.NO_PERMISSION.value)`

**Comment by CarlBarks** [comment]
We are experiencing the same issue:

```
Unable to create application xxx: org.json.JSONException: No value for notification_types

Caused by org.json.JSONException
No value for notification_types

com.onesignal.internal.OneSignalImp.initWithContext (OneSignalImp.kt:280)
```

We are on OneSIgnal version `5.1.1`

So far 2 crashes on a `Model:Galaxy S8` on `Android 9`

**Comment by area51bis** [comment]
As workaround, you can execute this code before initializing OneSignal:

```kotlin
private fun fixMissingNotificationTypes(context: Context) {
    val prefs: SharedPreferences = context.getSharedPreferences(PreferenceStores.ONESIGNAL, Context.MODE_PRIVATE)
    val legacyUserSyncString = prefs.getString(PreferenceOneSignalKeys.PREFS_LEGACY_USER_SYNCVALUES, null)
    if (legacyUserSyncString != null) {
        val legacyUserSyncJSON = JSONObject(legacyUserSyncString)
        // in some cases this field does not exist
        if (!legacyUserSyncJSON.has("notification_types")) {
            // set to 0 (NO_PERMISSION)
            legacyUserSyncJSON.put("notification_types", SubscriptionStatus.NO_PERMISSION.value)
            prefs.edit {
                putString(PreferenceOneSignalKeys.PREFS_LEGACY_USER_SYNCVALUES, legacyUserSyncJSON.toString())
            }
        }
    }
}
```

This load the OneSignal preferences and set "notification_types" to NO_PERMISSION if it doesn't exists.

**Comment by jinliu9508** [comment]
@area51bis Thank you for your response! I believe your suggestion on the fix and workaround is very precise. We are working on a fix thanks to your input.

**Comment by jinliu9508** [comment]
Fixed in 5.1.4.

**Comment by area51bis** [comment]
Thanks!
